src/renderer/state.ts

```diff
@@ -208,6 +208,34 @@ export class AppState {
     versions: this.baseVersions,
   });
 
+  // Lock on the active Electron version that prevents other windows from removing it
+  private versionLock: Lock | null = null;
+
+  // Used to release the lock when the current window switches Electron versions
+  private versionLockController = new AbortController();
+
+  private static versionLockNamePrefix = 'version:';
```

Comment:
Part of me thinks this should go in `src/renderer/constants.ts`, but I could go either way, I don't feel strongly.